lib/inspec/profile.rb

```diff
@@ -94,6 +94,7 @@ def initialize(source_reader, options = {})
       @input_values = options[:inputs]
       @tests_collected = false
       @libraries_loaded = false
+      @failed = false
```

Comment:
I think you're right that having `failed` as an instance variable might not be the right place.
However, I'm not seeing an existing `status` in the metadata. The `skipped` status is instead inferred by the profile by calling [`supports_platform?`](https://github.com/inspec/inspec/blob/master/lib/inspec/profile.rb#L327).

How would you see this new `failed` property neatly fitting into the metadata? It could be a new `status` metadata property, but I'd need to rework how skipped profiles work as well.

Reply:
This is an awkward location, but there is a status on the dependency structure, which is set in a really weird way. 
https://github.com/inspec/inspec/pull/5128/files#diff-d1eede4085683a605d9b8892a198f642R265

Reply:
I did see that, but that is metadata for dependency profiles, not metadata for the immediate profile, no? Or can it be used to represent the metadata for the current profile?
At this point I'd rather just have the instance variable than a metadata entry that behaves weirdly.

Reply:
I agree the existing approach on Metadata's dependency structure is inappropriate.  

What I'm saying is that the responsibility for knowing the status of the profile is currently on the Metadata object. This PR now divides that responsibility between the Profile and Metadata objects, which creates a lot of potential for bugs.  An implementation could be chosen - an instance variable on Metadata, perhaps - that keeps the responsibility on one class. 

You could still have `Profile#failed?` - it can just call into metadata.

Reply:
I'll have a go at this, but I might need to leave a backlogged issue to fix how skipping works so that it is also a real status and not inferred indirectly.

Reply:
Looking at this a second time, `Metadata` is intended to represent the static metadata stored in `my_inspec_profile/metadata.rb`

https://github.com/inspec/inspec/blob/master/lib/inspec/metadata.rb#L12

It does not appear to be the right place to represent an instance of the profile loaded by the runner - this file (`lib/inspec/profile.rb`) is the correct place for that.

So I would like to keep the change here. What should change is to have the instance variable be `@state` not `@failed` and have it store whether this instance of the profile is `:loaded` `:failed` or `:skipped` (the latter would mean directly storing here if this instance of the profile is skipped instead of calling into the metadata).

@clintoncwolfe if you don't agree with me, can we at least agree on a refactoring issue to get the responsibility of Metadata fixed up and documented properly outside of this PR?

Reply:
Your approach sounds reasonable - having an ivar named state on Profile seems fine, and (ideally) moving the skipped status to be indicated using that ivar sounds good. Adding a comment to the effect that Metadata represents metadata and Profile represents state might be helpful, as they have some overlap. Up to you if you want that on a separate PR.

Reply:
Just had a look at this and appreciate the thought gone into this issue on the backscroll. Looking at this from a newer set of eyes I do appreciate the clarifying comment on metadata (on an initial reading of the object class my interpretation would have been that it solely holds responsibility for `metadata.rb`). 

I also found it useful to clarify that `Profile` is the runtime state representation, and the use of `@state` seems fitting to me. (Naming this is hard, no?) Thanks for pulling this together.